pkg/storage/minio/saver.go

```diff
@@ -7,28 +7,24 @@ import (
 
 	"github.com/gomods/athens/pkg/errors"
 	"github.com/gomods/athens/pkg/observ"
+	moduploader "github.com/gomods/athens/pkg/storage/module"
 	minio "github.com/minio/minio-go"
 )
 
-func (s *storageImpl) Save(ctx context.Context, module, vsn string, mod []byte, zip io.Reader, info []byte) error {
-	const op errors.Op = "storage.minio.Save"
+func (s *storageImpl) Save(ctx context.Context, module, version string, mod []byte, zip io.Reader, info []byte) error {
+	const op errors.Op = "minio.Save"
 	ctx, span := observ.StartSpan(ctx, op.String())
 	defer span.End()
-	dir := s.versionLocation(module, vsn)
-	modFileName := dir + "/" + "go.mod"
-	zipFileName := dir + "/" + "source.zip"
-	infoFileName := dir + "/" + vsn + ".info"
-	_, err := s.minioClient.PutObject(s.bucketName, modFileName, bytes.NewReader(mod), int64(len(mod)), minio.PutObjectOptions{})
-	if err != nil {
-		return errors.E(op, err)
-	}
-	_, err = s.minioClient.PutObject(s.bucketName, zipFileName, zip, -1, minio.PutObjectOptions{})
-	if err != nil {
-		return errors.E(op, err)
-	}
-	_, err = s.minioClient.PutObject(s.bucketName, infoFileName, bytes.NewReader(info), int64(len(info)), minio.PutObjectOptions{})
+
+	err := moduploader.Upload(ctx, module, version, bytes.NewReader(info), bytes.NewReader(mod), zip, s.upload, 300)
```

Comment:
We should use the value from the config here - time.Duration in sec. I think the test is failing b/c of that.

Reply:
Yep! that was one reason, the entire storage module will need to be updated to get this PR in. Also, this PR is a breaking change, so we should consider if we should really pull the trigger or not.